src/ursa/agents/rag_agent.py

```diff
@@ -0,0 +1,267 @@
+        embedding=None,
+        return_k: int = 10,
+        chunk_size=1000,
+        chunk_overlap=200,
+        database_path="database",
+        summaries_path="database",
+        vectorstore_path="vectorstore",
+        **kwargs,
+    ):
+        super().__init__(llm, **kwargs)
+        self.retriever = None
+        self._vs_lock = Lock()
+        self.return_k = return_k
+        self.embedding = embedding
+        self.chunk_size = chunk_size
+        self.chunk_overlap = chunk_overlap
+        self.database_path = database_path
+        self.summaries_path = summaries_path
+        self.vectorstore_path = vectorstore_path
+        self.graph = self._build_graph()
+
+        os.makedirs(self.vectorstore_path, exist_ok=True)
+        self.vectorstore = self._open_global_vectorstore()
+
+    def _open_global_vectorstore(self) -> Chroma:
+        return Chroma(
+            persist_directory=self.vectorstore_path,
+            embedding_function=self.embedding,
+        )
+
+    def _paper_exists_in_vectorstore(self, doc_id: str) -> bool:
+        try:
+            col = self.vectorstore._collection
+            res = col.get(where={"id": doc_id}, limit=1)
+            return len(res.get("ids", [])) > 0
+        except Exception:
+            manifest_path = os.path.join(
+                self.vectorstore_path, "_ingested_ids.txt"
+            )
+            if not os.path.exists(manifest_path):
+                return False
+            with open(manifest_path, "r") as f:
+                return any(line.strip() == doc_id for line in f)
+
+    def _mark_paper_ingested(self, arxiv_id: str) -> None:
+        manifest_path = os.path.join(self.vectorstore_path, "_ingested_ids.txt")
+        with open(manifest_path, "a") as f:
+            f.write(f"{arxiv_id}\n")
+
+    def _ensure_doc_in_vectorstore(self, paper_text: str, doc_id: str) -> None:
+        splitter = RecursiveCharacterTextSplitter(
+            chunk_size=self.chunk_size, chunk_overlap=self.chunk_overlap
+        )
+        docs = splitter.create_documents(
+            [paper_text], metadatas=[{"id": doc_id}]
+        )
+        with self._vs_lock:
+            if not self._paper_exists_in_vectorstore(doc_id):
+                base = f"{doc_id}::"
+                ids = [f"{base}{i}" for i in range(len(docs))]
+                self.vectorstore.add_documents(docs, ids=ids)
+                self._mark_paper_ingested(doc_id)
+
+    def _get_global_retriever(self, k: int = 5):
+        return self.vectorstore, self.vectorstore.as_retriever(
+            search_kwargs={"k": k}
+        )
+
+    def _read_docs(self, state: RAGState) -> RAGState:
+        print("[RAG Agent] Reading Documents....")
+        papers = []
+        new_state = state.copy()
+
+        pdf_files = [
+            f
+            for f in os.listdir(self.database_path)
+            if f.lower().endswith(".pdf")
+        ]
+
+        doc_ids = [pdf_filename.split(".pdf")[0] for pdf_filename in pdf_files]
```

Comment:
```suggestion
        doc_ids = [pdf_filename.rsplit(".pdf")[0] for pdf_filename in pdf_files]
```

Because this may happen:
```python
filename = "a.pdffoo.pdf"
filename.split(".pdf")  # ['a', 'foo.pdf']
filename.rsplit(".pdf", 1)  # ['a.pdffoo', ''], which is what you want.
```